File: src/dbflute/adf/meta/vendorLargeDataRefdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type VendorLargeDataRefDbm_T struct {
	df.BaseDBMeta
	ColumnLargeDataRefId *df.ColumnInfo
	ColumnLargeDataId *df.ColumnInfo
	ColumnDateIndex *df.ColumnInfo
	ColumnDateNoIndex *df.ColumnInfo
	ColumnTimestampIndex *df.ColumnInfo
	ColumnTimestampNoIndex *df.ColumnInfo
	ColumnNullableDecimalIndex *df.ColumnInfo
	ColumnNullableDecimalNoIndex *df.ColumnInfo
	ColumnSelfParentId *df.ColumnInfo
}

func (b *VendorLargeDataRefDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *VendorLargeDataRefDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var VendorLargeDataRefDbm *VendorLargeDataRefDbm_T

func Create_VendorLargeDataRefDbm() {
	VendorLargeDataRefDbm = new(VendorLargeDataRefDbm_T)
	VendorLargeDataRefDbm.TableDbName = "vendor_large_data_ref"
	VendorLargeDataRefDbm.TableDispName = "vendor_large_data_ref"
	VendorLargeDataRefDbm.TablePropertyName = "vendorLargeDataRef"
	VendorLargeDataRefDbm.TableSqlName = new(df.TableSqlName)
	VendorLargeDataRefDbm.TableSqlName.TableSqlName = "vendor_large_data_ref"
	VendorLargeDataRefDbm.TableSqlName.CorrespondingDbName = VendorLargeDataRefDbm.TableDbName

	var vendorLargeDataRef df.DBMeta
	vendorLargeDataRef = VendorLargeDataRefDbm
	VendorLargeDataRefDbm.DBMeta=&vendorLargeDataRef
	largeDataRefIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo large_data_ref_id
	largeDataRefIdSqlName.ColumnSqlName = "large_data_ref_id"
	largeDataRefIdSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnLargeDataRefId = df.CCI(&vendorLargeDataRef, "large_data_ref_id", largeDataRefIdSqlName, "", "", "Long.class", "largeDataRefId", "", true, false,true, "int8", 19, 0, "",false,"","", "","vendorLargeDataRefSelfList","",false,"int64")
	largeDataIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo large_data_id
	largeDataIdSqlName.ColumnSqlName = "large_data_id"
	largeDataIdSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnLargeDataId = df.CCI(&vendorLargeDataRef, "large_data_id", largeDataIdSqlName, "", "", "Long.class", "largeDataId", "", false, false,true, "int8", 19, 0, "",false,"","", "vendorLargeData","","",false,"int64")
	dateIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo date_index
	dateIndexSqlName.ColumnSqlName = "date_index"
	dateIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnDateIndex = df.CCI(&vendorLargeDataRef, "date_index", dateIndexSqlName, "", "", "java.time.LocalDate.class", "dateIndex", "", false, false,true, "date", 13, 0, "",false,"","", "","","",false,"df.Date")
	dateNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo date_no_index
	dateNoIndexSqlName.ColumnSqlName = "date_no_index"
	dateNoIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnDateNoIndex = df.CCI(&vendorLargeDataRef, "date_no_index", dateNoIndexSqlName, "", "", "java.time.LocalDate.class", "dateNoIndex", "", false, false,true, "date", 13, 0, "",false,"","", "","","",false,"df.Date")
	timestampIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo timestamp_index
	timestampIndexSqlName.ColumnSqlName = "timestamp_index"
	timestampIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnTimestampIndex = df.CCI(&vendorLargeDataRef, "timestamp_index", timestampIndexSqlName, "", "", "java.time.LocalDateTime.class", "timestampIndex", "", false, false,true, "timestamp", 26, 3, "",false,"","", "","","",false,"df.Timestamp")
	timestampNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo timestamp_no_index
	timestampNoIndexSqlName.ColumnSqlName = "timestamp_no_index"
	timestampNoIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnTimestampNoIndex = df.CCI(&vendorLargeDataRef, "timestamp_no_index", timestampNoIndexSqlName, "", "", "java.time.LocalDateTime.class", "timestampNoIndex", "", false, false,true, "timestamp", 26, 3, "",false,"","", "","","",false,"df.Timestamp")
	nullableDecimalIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo nullable_decimal_index
	nullableDecimalIndexSqlName.ColumnSqlName = "nullable_decimal_index"
	nullableDecimalIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnNullableDecimalIndex = df.CCI(&vendorLargeDataRef, "nullable_decimal_index", nullableDecimalIndexSqlName, "", "", "java.math.BigDecimal.class", "nullableDecimalIndex", "", false, false,false, "numeric", 12, 3, "",false,"","", "","","",false,"df.NullNumeric")
	nullableDecimalNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo nullable_decimal_no_index
	nullableDecimalNoIndexSqlName.ColumnSqlName = "nullable_decimal_no_index"
	nullableDecimalNoIndexSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnNullableDecimalNoIndex = df.CCI(&vendorLargeDataRef, "nullable_decimal_no_index", nullableDecimalNoIndexSqlName, "", "", "java.math.BigDecimal.class", "nullableDecimalNoIndex", "", false, false,false, "numeric", 12, 3, "",false,"","", "","","",false,"df.NullNumeric")
	selfParentIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo self_parent_id
	selfParentIdSqlName.ColumnSqlName = "self_parent_id"
	selfParentIdSqlName.IrregularChar = false
	VendorLargeDataRefDbm.ColumnSelfParentId = df.CCI(&vendorLargeDataRef, "self_parent_id", selfParentIdSqlName, "", "", "Long.class", "selfParentId", "", false, false,false, "int8", 19, 0, "",false,"","", "vendorLargeDataRefSelf","","",false,"sql.NullInt64")

	VendorLargeDataRefDbm.ColumnInfoList = new(df.List)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnLargeDataRefId)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnLargeDataId)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnDateIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnDateNoIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnTimestampIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnTimestampNoIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnNullableDecimalIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnNullableDecimalNoIndex)
	VendorLargeDataRefDbm.ColumnInfoList.Add(VendorLargeDataRefDbm.ColumnSelfParentId)


	VendorLargeDataRefDbm.ColumnInfoMap=make(map[string]int)
	VendorLargeDataRefDbm.ColumnInfoMap["largeDataRefId"]=0
		VendorLargeDataRefDbm.ColumnInfoMap["largeDataId"]=1
		VendorLargeDataRefDbm.ColumnInfoMap["dateIndex"]=2
		VendorLargeDataRefDbm.ColumnInfoMap["dateNoIndex"]=3
		VendorLargeDataRefDbm.ColumnInfoMap["timestampIndex"]=4
		VendorLargeDataRefDbm.ColumnInfoMap["timestampNoIndex"]=5
		VendorLargeDataRefDbm.ColumnInfoMap["nullableDecimalIndex"]=6
		VendorLargeDataRefDbm.ColumnInfoMap["nullableDecimalNoIndex"]=7
		VendorLargeDataRefDbm.ColumnInfoMap["selfParentId"]=8
	    VendorLargeDataRefDbm.PrimaryKey = true
    VendorLargeDataRefDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &vendorLargeDataRef
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(VendorLargeDataRefDbm.ColumnLargeDataRefId)

	VendorLargeDataRefDbm.PrimaryInfo = new(df.PrimaryInfo)
	VendorLargeDataRefDbm.PrimaryInfo.UniqueInfo = ui
	
	var vendorLargeDataRefMeta df.DBMeta = VendorLargeDataRefDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["VendorLargeDataRef"] = &vendorLargeDataRefMeta
}

File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/VendorLargeDataRefDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.optional.OptionalEntity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of vendor_large_data_ref. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class VendorLargeDataRefDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final VendorLargeDataRefDbm _instance = new VendorLargeDataRefDbm();
    private VendorLargeDataRefDbm() {}
    public static VendorLargeDataRefDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getLargeDataRefId(), (et, vl) -> ((VendorLargeDataRef)et).setLargeDataRefId(ctl(vl)), "largeDataRefId");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getLargeDataId(), (et, vl) -> ((VendorLargeDataRef)et).setLargeDataId(ctl(vl)), "largeDataId");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getDateIndex(), (et, vl) -> ((VendorLargeDataRef)et).setDateIndex(ctld(vl)), "dateIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getDateNoIndex(), (et, vl) -> ((VendorLargeDataRef)et).setDateNoIndex(ctld(vl)), "dateNoIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getTimestampIndex(), (et, vl) -> ((VendorLargeDataRef)et).setTimestampIndex(ctldt(vl)), "timestampIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getTimestampNoIndex(), (et, vl) -> ((VendorLargeDataRef)et).setTimestampNoIndex(ctldt(vl)), "timestampNoIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getNullableDecimalIndex(), (et, vl) -> ((VendorLargeDataRef)et).setNullableDecimalIndex(ctb(vl)), "nullableDecimalIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getNullableDecimalNoIndex(), (et, vl) -> ((VendorLargeDataRef)et).setNullableDecimalNoIndex(ctb(vl)), "nullableDecimalNoIndex");
        setupEpg(_epgMap, et -> ((VendorLargeDataRef)et).getSelfParentId(), (et, vl) -> ((VendorLargeDataRef)et).setSelfParentId(ctl(vl)), "selfParentId");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    protected final Map<String, PropertyGateway> _efpgMap = newHashMap();
    { xsetupEfpg(); }
    @SuppressWarnings("unchecked")
    protected void xsetupEfpg() {
        setupEfpg(_efpgMap, et -> ((VendorLargeDataRef)et).getVendorLargeData(), (et, vl) -> ((VendorLargeDataRef)et).setVendorLargeData((OptionalEntity<VendorLargeData>)vl), "vendorLargeData");
        setupEfpg(_efpgMap, et -> ((VendorLargeDataRef)et).getVendorLargeDataRefSelf(), (et, vl) -> ((VendorLargeDataRef)et).setVendorLargeDataRefSelf((OptionalEntity<VendorLargeDataRef>)vl), "vendorLargeDataRefSelf");
    }
    public PropertyGateway findForeignPropertyGateway(String prop)
    { return doFindEfpg(_efpgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "vendor_large_data_ref";
    protected final String _tableDispName = "vendor_large_data_ref";
    protected final String _tablePropertyName = "vendorLargeDataRef";
    protected final TableSqlName _tableSqlName = new TableSqlName("vendor_large_data_ref", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnLargeDataRefId = cci("large_data_ref_id", "large_data_ref_id", null, null, Long.class, "largeDataRefId", null, true, false, true, "int8", 19, 0, null, false, null, null, null, "vendorLargeDataRefSelfList", null, false);
//"int64"
    protected final ColumnInfo _columnLargeDataId = cci("large_data_id", "large_data_id", null, null, Long.class, "largeDataId", null, false, false, true, "int8", 19, 0, null, false, null, null, "vendorLargeData", null, null, false);
//"int64"
    protected final ColumnInfo _columnDateIndex = cci("date_index", "date_index", null, null, java.time.LocalDate.class, "dateIndex", null, false, false, true, "date", 13, 0, null, false, null, null, null, null, null, false);
//"df.Date"
    protected final ColumnInfo _columnDateNoIndex = cci("date_no_index", "date_no_index", null, null, java.time.LocalDate.class, "dateNoIndex", null, false, false, true, "date", 13, 0, null, false, null, null, null, null, null, false);
//"df.Date"
    protected final ColumnInfo _columnTimestampIndex = cci("timestamp_index", "timestamp_index", null, null, java.time.LocalDateTime.class, "timestampIndex", null, false, false, true, "timestamp", 26, 3, null, false, null, null, null, null, null, false);
//"df.Timestamp"
    protected final ColumnInfo _columnTimestampNoIndex = cci("timestamp_no_index", "timestamp_no_index", null, null, java.time.LocalDateTime.class, "timestampNoIndex", null, false, false, true, "timestamp", 26, 3, null, false, null, null, null, null, null, false);
//"df.Timestamp"
    protected final ColumnInfo _columnNullableDecimalIndex = cci("nullable_decimal_index", "nullable_decimal_index", null, null, java.math.BigDecimal.class, "nullableDecimalIndex", null, false, false, false, "numeric", 12, 3, null, false, null, null, null, null, null, false);
//"df.NullNumeric"
    protected final ColumnInfo _columnNullableDecimalNoIndex = cci("nullable_decimal_no_index", "nullable_decimal_no_index", null, null, java.math.BigDecimal.class, "nullableDecimalNoIndex", null, false, false, false, "numeric", 12, 3, null, false, null, null, null, null, null, false);
//"df.NullNumeric"
    protected final ColumnInfo _columnSelfParentId = cci("self_parent_id", "self_parent_id", null, null, Long.class, "selfParentId", null, false, false, false, "int8", 19, 0, null, false, null, null, "vendorLargeDataRefSelf", null, null, false);
//"sql.NullInt64"

    /**
     * large_data_ref_id: {PK, NotNull, int8(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnLargeDataRefId() { return _columnLargeDataRefId; }
    /**
     * large_data_id: {NotNull, int8(19), FK to vendor_large_data}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnLargeDataId() { return _columnLargeDataId; }
    /**
     * date_index: {IX, NotNull, date(13)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDateIndex() { return _columnDateIndex; }
    /**
     * date_no_index: {NotNull, date(13)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDateNoIndex() { return _columnDateNoIndex; }
    /**
     * timestamp_index: {IX, NotNull, timestamp(26, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnTimestampIndex() { return _columnTimestampIndex; }
    /**
     * timestamp_no_index: {NotNull, timestamp(26, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnTimestampNoIndex() { return _columnTimestampNoIndex; }
    /**
     * nullable_decimal_index: {IX, numeric(12, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNullableDecimalIndex() { return _columnNullableDecimalIndex; }
    /**
     * nullable_decimal_no_index: {numeric(12, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNullableDecimalNoIndex() { return _columnNullableDecimalNoIndex; }
    /**
     * self_parent_id: {int8(19), FK to vendor_large_data_ref}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnSelfParentId() { return _columnSelfParentId; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnLargeDataRefId());
        ls.add(columnLargeDataId());
        ls.add(columnDateIndex());
        ls.add(columnDateNoIndex());
        ls.add(columnTimestampIndex());
        ls.add(columnTimestampNoIndex());
        ls.add(columnNullableDecimalIndex());
        ls.add(columnNullableDecimalNoIndex());
        ls.add(columnSelfParentId());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnLargeDataRefId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    /**
     * vendor_large_data by my large_data_id, named 'vendorLargeData'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignVendorLargeData() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnLargeDataId(), VendorLargeDataDbm.getInstance().columnLargeDataId());
        return cfi("fk_vendor_large_data_ref_data", "vendorLargeData", this, VendorLargeDataDbm.getInstance(), mp, 0, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "vendorLargeDataRefList", false);
    }
    /**
     * vendor_large_data_ref by my self_parent_id, named 'vendorLargeDataRefSelf'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignVendorLargeDataRefSelf() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnSelfParentId(), VendorLargeDataRefDbm.getInstance().columnLargeDataRefId());
        return cfi("fk_vendor_large_data_ref_self", "vendorLargeDataRefSelf", this, VendorLargeDataRefDbm.getInstance(), mp, 1, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "vendorLargeDataRefSelfList", false);
    }

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------
    /**
     * vendor_large_data_ref by self_parent_id, named 'vendorLargeDataRefSelfList'.
     * @return The information object of referrer property. (NotNull)
     */
    public ReferrerInfo referrerVendorLargeDataRefSelfList() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnLargeDataRefId(), VendorLargeDataRefDbm.getInstance().columnSelfParentId());
        return cri("fk_vendor_large_data_ref_self", "vendorLargeDataRefSelfList", this, VendorLargeDataRefDbm.getInstance(), mp, false, "vendorLargeDataRefSelf");
    }

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.VendorLargeDataRef"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.VendorLargeDataRefCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.VendorLargeDataRefBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<VendorLargeDataRef> getEntityType() { return VendorLargeDataRef.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public VendorLargeDataRef newEntity() { return new VendorLargeDataRef(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((VendorLargeDataRef)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((VendorLargeDataRef)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}
